core/domain/exp_domain.py

```diff
@@ -2187,6 +2217,67 @@ def _convert_states_v28_dict_to_v29_dict(cls, states_dict):
             state_dict['solicit_answer_details'] = False
         return states_dict
 
+    @classmethod
+    def _convert_states_v29_dict_to_v30_dict(cls, states_dict):
+        """Converts from version 29 to 30. Version 30 added image assets field
+        in state model. This funtion is extracting HTMl of state and do 3
+        operations on each HTML content. The 3 operations and their sequence
+        were.
+        1) Gets image src of all images present in a state.
+        2) Assign each image source an image id or say maps each image id with
```

Comment:
Skip the part before or
```suggestion
        2) Maps each image with an image ID.
```

Reply:
done